### PR TITLE
feat(training): add GR00T VLA fine-tuning workflow for OSMO

### PR DESCRIPTION
**Pull request by akhanattentive**
# Pull Request

## Description

Adds GR00T-N1.5 / N1.7 vision-language-action fine-tuning as a first-class training path on OSMO, alongside the existing LeRobot ACT / Diffusion flows. Introduces a dedicated submission script, OSMO workflow, embodiment-agnostic config templates, and documentation covering both GR00T codebase versions.

**What's included:**

- `training/vla/scripts/submit-osmo-lerobot-vla-fine-tuning.sh` — submission CLI with `--vla-version {1.5,1.7}`, dataset blob URL injection, auto-resolution of data/modality config files, optional Azure ML checkpoint mirror, optional ACR OCI push, and resume support.
- `training/vla/workflows/osmo/groot-train.yaml` — self-contained OSMO workflow that clones Isaac-GR00T at the version-pinned ref, materializes the data/modality config from env-injected base64 payloads, and runs fine-tuning on `h100gpu_platform`.
- `training/vla/configs/groot/examples/` — embodiment-agnostic reference templates: `data_config.py` (N1.5 `DATA_CONFIG_MAP["example"]`), `modality_config.py` (N1.7 `ModalityConfig`), and a README with adapt-for-your-embodiment guidance.
- `docs/training/lerobot-training.md` — adds GR00T-N1.7 row to the policy architectures table, renames the GR00T section to cover both N1.5/N1.7, splits the Quick Start into per-version examples, and extends the GR00T parameter table with `--vla-version`, `--data-config-file`, `--modality-config-file`, and `--acr-registry`.
- `docs/training/osmo-training.md` — adds a `🤖 GR00T VLA Fine-Tuning` section with per-version submit examples and cross-links to the lerobot doc.
- `training/vla/README.md` — directory tree, Quick Start, config layout.
- `pyproject.toml` — per-file-ignore for the example `data_config.py` fragment (F821, RUF012 — file references symbols only resolvable after runtime append).

Closes #

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change fixing an issue)
- [x] ✨ New feature (non-breaking change adding functionality)
- [ ] 💥 Breaking change (fix or feature causing existing functionality to change)
- [x] 📚 Documentation update
- [ ] 🏗️ Infrastructure change (Terraform/IaC)
- [ ] ♻️ Refactoring (no functional changes)

## Component(s) Affected

- [ ] `infrastructure/terraform/prerequisites/` - Azure subscription setup
- [ ] `infrastructure/terraform/` - Terraform infrastructure
- [ ] `infrastructure/setup/` - OSMO control plane / Helm
- [x] `workflows/` - Training and evaluation workflows
- [x] `training/` - Training pipelines and scripts
- [x] `docs/` - Documentation

## Testing Performed

- [ ] Terraform `plan` reviewed (no unexpected changes)
- [ ] Terraform `apply` tested in dev environment
- [ ] Training scripts tested locally with Isaac Sim
- [x] OSMO workflow submitted successfully
- [ ] Smoke tests passed (`smoke_test_azure.py`)

Additional validation:

- `shellcheck training/vla/scripts/submit-osmo-lerobot-vla-fine-tuning.sh` — clean
- `ruff check training/vla/` — clean (with new per-file-ignore for the example fragment)
- `npm run lint:md` — clean for all touched docs
- OSMO workflow submitted and validated for both GR00T-N1.5 and N1.7 against the example data config

## Documentation Impact

- [ ] No documentation changes needed
- [x] Documentation updated in this PR
- [ ] Documentation issue filed

## Bug Fix Checklist

*Complete this section for bug fix PRs. Skip for other contribution types.*

- [ ] Linked to issue being fixed
- [ ] Regression test included, OR
- [ ] Justification for no regression test:

## Checklist

- [x] My code follows the [project conventions](copilot-instructions.md)
- [x] Commit messages follow [conventional commit format](instructions/commit-message.instructions.md)
- [x] I have performed a self-review
- [x] Documentation impact assessed above
- [x] No new linting warnings introduced

### CODE REVIEW

**Review by bindsi — COMMENTED** [review]
Automated review (draft PR): Quick scan — GR00T VLA fine-tuning workflow addition looks well-structured (cspell additions, documentation, workflow YAML). No blocking issues found on initial review. Will re-review when ready for merge.

File: training/vla/workflows/osmo/groot-train.yaml [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+            endpoint = f"https://{parsed.netloc}"
+            sas_token = parsed.query if parsed.query else None
+
+            def download_with_client(client: ContainerClient) -> int:
+              count = 0
+              for blob in client.list_blobs(name_starts_with=prefix):
+                rel = blob.name[len(prefix):].lstrip("/")
+                if not rel:
+                  continue
+                local = f"/data/dataset/{rel}"
```
Comment (bindsi) [inline training/vla/workflows/osmo/groot-train.yaml]: The blob downloader writes `local = f"/data/dataset/{rel}"` without normalizing and verifying the result stays under `/data/dataset`. Azure blob names can contain `../`, so a malicious blob under the selected prefix could write outside the dataset directory. Please resolve the path under the dataset root and reject any path that escapes it before creating directories or opening the file.

**Review by bindsi — CHANGES_REQUESTED** [review]
Automated batch review: requesting changes for one blocking path traversal issue.

**Review by bindsi — COMMENTED** [review]
Automated batch re-review: non-blocking findings for follow-up.

1. The PR body has an empty `Closes #` value, so issue alignment and scope cannot be verified.
2. `training/vla/scripts/submit-osmo-lerobot-vla-fine-tuning.sh` validates required args but not `--blob-url`, although the workflow needs a dataset source; please fail fast for a missing blob URL.
3. `training/vla/workflows/osmo/groot-train.yaml` hardcodes `DATASET_PATH`, ignoring the submitted `dataset_path`.
4. `training/vla/workflows/osmo/groot-train.yaml` passes an ACR refresh token via `oras login --password`; use `--password-stdin` instead.
5. `training/vla/configs/groot/examples/modality_config.py` is missing the repo-standard first import `from __future__ import annotations`.

**Review by bindsi — COMMENTED** [review]
Re-review after the `main` merge: no new blockers. The GR00T VLA fine-tuning path (submission script, OSMO workflow, config templates, and docs) is consistent, and the reported `shellcheck`/`ruff`/`lint:md` runs came back clean.

The previously raised non-blocking items still appear to apply (they don't look addressed by the merge) — please follow up on those when you get a chance. Nothing here blocks merge from my side.

**Comment by akhanattentive** [comment]
@bindsi did the feedback above. Retested osmo workflow and end to end flow and still works 

The four fixes:

1. Missing --blob-url now fails fast: [ERROR] --blob-url is required (no dataset source configured) (shellcheck passes — no output).

2. dataset_path honored: DATASET_PATH: "{{dataset_path}}" env, download_blob.py derives DATASET_ROOT from the env var, and default-values ships dataset_path: "/data/dataset".

3. oras login --password-stdin: refresh token is now piped via printf '%s' | oras ... --password-stdin and unset after login.

4. from __future__ import annotations added to standalone modality_config.py (data_config.py intentionally omits it — it is appended into an existing file via >>, where future imports mid-file would be a SyntaxError).

**Review by bindsi — APPROVED** [review]
Approved: the GR00T VLA fine-tuning workflow changes are consistent and the previously noted items remain non-blocking. I did not find actionable blockers in the current diff.

File: training/vla/workflows/osmo/groot-train.yaml [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+                raise SystemExit(0)
+              except HttpResponseError as err:
+                if getattr(err, "error_code", None) not in {"AuthorizationFailure", "AuthenticationFailed"}:
+                  raise
+                print("Credentialed blob access failed; retrying anonymous access")
+
+            anon_client = ContainerClient(endpoint, container)
+            count = download_with_client(anon_client)
+            print(f"Downloaded {count} files to {DATASET_ROOT}/")
+        - path: /tmp/mlflow_log.py
```
Comment (akzaidi) [inline training/vla/workflows/osmo/groot-train.yaml:145]: I wonder if we can cover this with [`training/utils/aml_mirror.py`](https://github.com/microsoft/physical-ai-toolchain/blob/956e7520472860388def20681c5c48cca3a27e4e/training/utils/aml_mirror.py#L73)? I realize we don't mount the repo during the OSMO jobs or make it installable but instead of maintaining multiple separate paths for the AML MLFlow logging maybe we can attach this directly as a file to the current path and reuse it?

**Review by akzaidi — APPROVED** [review]

**Review by bindsi — APPROVED** [review]
Approved: the GR00T VLA fine-tuning workflow changes are consistent after re-review, including path and blob URL validation. I did not find actionable blockers.
